### PR TITLE
Refactor RBAC authorizer entry points

### PR DESCRIPTION
**Pull request by enj**
This change refactors various RBAC authorizer functions to be more flexible in their inputs.  This makes it easier to reuse the various components that make up the authorizer.

Signed-off-by: Monis Khan <mkhan@redhat.com>

```release-note
NONE
```

### CODE REVIEW

**Comment by enj** [comment]
@kubernetes/sig-auth-pr-reviews 
/sig auth
/assign @deads2k

File: plugin/pkg/auth/authorizer/rbac/rbac.go [hunk 1]
```diff
@@ -121,13 +120,8 @@ func (r *RBACAuthorizer) Authorize(requestAttributes authorizer.Attributes) (boo
...
 }
 
-func New(roles rbacregistryvalidation.RoleGetter, roleBindings rbacregistryvalidation.RoleBindingLister, clusterRoles rbacregistryvalidation.ClusterRoleGetter, clusterRoleBindings rbacregistryvalidation.ClusterRoleBindingLister) *RBACAuthorizer {
-	authorizer := &RBACAuthorizer{
-		authorizationRuleResolver: rbacregistryvalidation.NewDefaultRuleResolver(
-			roles, roleBindings, clusterRoles, clusterRoleBindings,
-		),
-	}
-	return authorizer
+func New(ruleResolver RequestToRuleMapper) *RBACAuthorizer {
```
Comment (deads2k) [inline plugin/pkg/auth/authorizer/rbac/rbac.go]: I don't like coercing the interfaces this way.  I do appreciate the difficulty in casting the listers (which you have) to the rbac interfaces (which you don't have), but I'd rather see the shims moved and be exposed here.

File: plugin/pkg/auth/authorizer/rbac/subject_locator.go [hunk 2]
```diff
@@ -39,16 +45,13 @@ type SubjectAccessEvaluator struct {
 	roleToRuleMapper         RoleToRuleMapper
 }
 
-func NewSubjectAccessEvaluator(roles rbacregistryvalidation.RoleGetter, roleBindings rbacregistryvalidation.RoleBindingLister, clusterRoles rbacregistryvalidation.ClusterRoleGetter, clusterRoleBindings rbacregistryvalidation.ClusterRoleBindingLister, superUser string) *SubjectAccessEvaluator {
-	subjectLocator := &SubjectAccessEvaluator{
+func NewSubjectAccessEvaluator(roleBindings rbacregistryvalidation.RoleBindingLister, clusterRoleBindings rbacregistryvalidation.ClusterRoleBindingLister, roleToRuleMapper RoleToRuleMapper, superUser string) *SubjectAccessEvaluator {
```
Comment (deads2k) [inline plugin/pkg/auth/authorizer/rbac/subject_locator.go]: I'd keep this signature as it was.

**Pull request merged**